### PR TITLE
Move ci-cleaner to Go modules

### PR DESCRIPTION
**Pull request by stone-z**
Migrate to Go modules. Toward https://github.com/giantswarm/giantswarm/issues/5680

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,24 @@
+module github.com/giantswarm/ci-cleaner
+
+go 1.13
+
+require (
+	github.com/Azure/azure-sdk-for-go v19.1.0+incompatible
```
Comment (MarcelMue) [inline go.mod]: Is it bad that it says incompatible? What does that mean exactly? :D

**Review by MarcelMue — APPROVED** [review]
LGTM but not really great at go modules.

File: go.mod (hunk 1 above)
Comment (sslavic) [reply go.mod]: AFAIK, library/module releases for v2+ should have /v2+ in their Go module path e.g. acme.org/foo/bar for v0.* or v1.* releases and `acme.org/foo/bar/v2+` for v2+ releases; if this convention is followed in a library, that enables library users to use multiple major versions of that library in same project. This concept is called semantic import versioning.

If library uses semver but its Go module path doesn't follow the semantic import versioning convention (e.g. library doesn't use Go modules at all like azure sdk), projects using Go modules which depend on that library will see that dependency marked with `+incompatible`. It's nothing bad, one just won't be able to simultaneously use multiple major versions of that library in a single project.

For great intro to Go modules, and pros/cons of this concept see https://github.com/liggitt/gomodules#major-versions

Official docs on this: 
- https://github.com/golang/go/wiki/Modules#semantic-import-versioning
- https://github.com/golang/go/wiki/Modules#can-a-module-consume-a-v2-package-that-has-not-opted-into-modules-what-does-incompatible-mean

**Review by kopiczko — DISMISSED** [review]
When it still compiles :)

**Review by kopiczko — APPROVED** [review]
`architect build` works with go modules so ignore my previous comment.

File: go.mod (hunk 1 above)
Comment (MarcelMue) [reply go.mod]: Thanks for the detailed explanation @sslavic - makes sense to me now :+1:

**Pull request merged**